lighthouse-core/audits/screenshot-thumbnails.js

```diff
@@ -0,0 +1,119 @@
+      description: 'Screenshot Thumbnails',
+      helpText: 'This is what the load of your site looked like.',
+      requiredArtifacts: ['traces']
+    };
+  }
+
+  /**
+   * Scales down an image to THUMBNAIL_WIDTH using nearest neighbor for speed, maintains aspect
+   * ratio of the original thumbnail.
+   *
+   * @param {{width: number, height: number, data: !Array<number>}} imageData
+   * @return {{width: number, height: number, data: !Array<number>}}
+   */
+  static scaleImageToThumbnail(imageData) {
+    const scaledWidth = THUMBNAIL_WIDTH;
+    const scaleFactor = imageData.width / scaledWidth;
+    const scaledHeight = Math.floor(imageData.height / scaleFactor);
+
+    const outPixels = new Uint8Array(scaledWidth * scaledHeight * 4);
+
+    for (let i = 0; i < scaledWidth; i++) {
+      for (let j = 0; j < scaledHeight; j++) {
+        const origX = Math.floor(i * scaleFactor);
+        const origY = Math.floor(j * scaleFactor);
+
+        const origPos = (origY * imageData.width + origX) * 4;
+        const outPos = (j * scaledWidth + i) * 4;
+
+        outPixels[outPos] = imageData.data[origPos];
+        outPixels[outPos + 1] = imageData.data[origPos + 1];
+        outPixels[outPos + 2] = imageData.data[origPos + 2];
+        outPixels[outPos + 3] = imageData.data[origPos + 3];
+      }
+    }
+
+    return {
+      width: scaledWidth,
+      height: scaledHeight,
+      data: outPixels,
+    };
+  }
+
+  /**
+   * @param {!Artifacts} artifacts
+   * @return {!AuditResult}
+   */
+  static audit(artifacts) {
+    const trace = artifacts.traces[Audit.DEFAULT_PASS];
+    const cachedThumbnails = new Map();
+
+    return artifacts.requestSpeedline(trace).then(speedline => {
+      const thumbnails = [];
+      const analyzedFrames = speedline.frames.filter(frame => !frame.isProgressInterpolated());
+
+      for (let i = 1; i <= NUMBER_OF_THUMBNAILS; i++) {
+        const targetTimestamp = speedline.beginning + speedline.complete * i / NUMBER_OF_THUMBNAILS;
+
+        let frameForTimestamp = null;
+        if (i === NUMBER_OF_THUMBNAILS) {
+          frameForTimestamp = analyzedFrames[analyzedFrames.length - 1];
+        } else {
+          analyzedFrames.forEach(frame => {
+            if (frame.getTimeStamp() <= targetTimestamp) {
+              frameForTimestamp = frame;
+            }
+          });
+        }
+
+        const imageData = frameForTimestamp.getParsedImage();
+        const thumbnailImageData = ScreenshotThumbnails.scaleImageToThumbnail(imageData);
+        const base64Data = cachedThumbnails.get(frameForTimestamp) ||
+            jpeg.encode(thumbnailImageData, 90).data.toString('base64');
+
+        cachedThumbnails.set(frameForTimestamp, base64Data);
+        thumbnails.push({
+          timing: Math.round(targetTimestamp - speedline.beginning),
+          timestamp: targetTimestamp * 1000,
+          data: base64Data,
+        });
+      }
+
+      return {rawValue: thumbnails};
```

Comment:
`rawValue` is completely inaccurately named, but it shouldn't be punished for that :)

> I wasn't a fan of those options originally since `details` maps to the details renderer and, again, rendering info not related to how to display the audit itself just its collapsible extraneous information.

I've actually been thinking about this distinction since #2140, and I increasingly think it actually makes sense to use `details` in this way.

tl;dr: the non-default category renderers already render audits however they want. In the same way, there's no reason why `details` can't be collapsed extra information *or* top level display information depending on what the particular category renderer chooses to display.

The performance category renderer is already bastardizing what the default category renderer does...e.g. any audit without a `group` is ignored completely, and with `screenshot-thumbnails` it plucks it out specially and renders it specially. The perf category renderer is pretty much inescapably special due to what we need from it. But really the method dispatch difference between `_renderDefaultCategory` and `_renderPerformanceCategory` is a lot larger than the conceptual difference.

`this._renderFilmstrip()` *is* a details renderer (conceptually), it's just that `renderAudit()` -> `renderAuditScore()` -> `detailsRenderer.render()` has been flattened into the few lines that call `renderFilmstrip()`. It's a `renderAuditScore()` with the top audit information hidden and the `details` rendered not in a `<details>` element. This could easily be made explicit with a `renderPerfAudit()` method that dispatches based on `group` or whatever, but it's probably not worth it :)

And putting the info in `details` means that this and future audits like this are compatible with other category renderers...if someone wants to use the `ScreenshotThumbnails` in their own custom report, it would work without them having to make their own perf category with all the right setup.